app/feature/feature-profile/src/main/kotlin/com/hedvig/android/feature/profile/payment/PaymentViewModel.kt

```diff
@@ -114,23 +114,25 @@ private fun PaymentData.toUiState(locale: Locale): PaymentViewModel.PaymentUiSta
         displayName = it.displayValue ?: "-",
       )
     },
-    paymentMethod = bankAccount?.let {
-      PaymentViewModel.PaymentUiState.PaymentMethod(
-        displayName = it.name,
-        displayValue = it.accountNumber,
+    paymentMethod = if (paymentMethod == null) {
+      null
+    } else {
+      bankAccount?.let {
+        PaymentViewModel.PaymentUiState.PaymentMethod(
+          displayName = it.name,
+          displayValue = it.accountNumber,
+        )
+      } ?: PaymentViewModel.PaymentUiState.PaymentMethod(
+        displayName = when (paymentMethod) {
+          is PaymentData.PaymentMethod.CardPaymentMethod -> paymentMethod.brand ?: "Unknown"
```

Comment:
Do we need "Unknown" localized?

Reply:
This ~should~ never happen so I think its ok to not have it localized. We can look at handling this better when we move the API to octopus